File: scripts/read-pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extractText(filePath) {
    try {
        const dataBuffer = fs.readFileSync(filePath);
        const data = await pdf(dataBuffer);
        console.log(`\n\n--- EXTRACTED: ${filePath} ---\n\n`);
        console.log(data.text);
    } catch (e) {
        console.error(`Error reading ${filePath}:`, e);
    }
}

async function main() {
    await extractText('C:\\Users\\SSFL-RETAIL-017\\Downloads\\enable - SEBIDeck_pvt circulation Dec 2025.pdf');
    await extractText('C:\\Users\\SSFL-RETAIL-017\\Downloads\\sebi circular dated -8th dec 2025.pdf');
}

main();

File: scripts/read_pdfs.py
import PyPDF2

files = [
    r"C:\Users\SSFL-RETAIL-017\Downloads\enable - SEBIDeck_pvt circulation Dec 2025.pdf",
    r"C:\Users\SSFL-RETAIL-017\Downloads\sebi circular dated -8th dec 2025.pdf"
]

with open(r"C:\Users\SSFL-RETAIL-017\sunidhi-nextjs\scripts\pdf-output.txt", "w", encoding="utf-8") as out_f:
    for file_path in files:
        out_f.write(f"\n\n--- EXTRACTED: {file_path} ---\n\n")
        try:
            with open(file_path, "rb") as pdf_f:
                reader = PyPDF2.PdfReader(pdf_f)
                for page in reader.pages:
                    text = page.extract_text()
                    if text:
                        out_f.write(text + "\n")
        except Exception as e:
            out_f.write(f"Error reading file: {e}\n")
